pkg/daemon/update.go

```diff
@@ -1809,6 +1828,41 @@ func (dn *Daemon) atomicallyWriteSSHKey(keys string) error {
 	return nil
 }
 
+// Set a given PasswdUser's Password Hash
+func (dn *Daemon) SetPasswordHash(newUsers []ign3types.PasswdUser) error {
+	// confirm that user exits
+	if len(newUsers) == 0 {
+		return nil
+	}
+
+	// mock function to mock the nodes existence  (it shouldnt run if no node exists)
+	var uErr user.UnknownUserError
+	switch _, err := user.Lookup(constants.CoreUserName); {
+	case err == nil:
+	case errors.As(err, &uErr):
+		glog.Info("core user does not exist, and creating users is not supported, so ignoring configuration specified for core user")
+		return nil
+	default:
+		return fmt.Errorf("failed to check if user core exists: %w", err)
+	}
+
+	// SetPasswordHash sets the password hash of the specified user.
+	for _, u := range newUsers {
+
+		pwhash := "*"
+		if *u.PasswordHash != "" || u.PasswordHash != nil {
```

Comment:
I think this is probably why we get the panic -- deref before nil check. 

I also think the line numbers probably don't match up to the panic because of how the test merges the code when it builds the PR: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/3539/pull-ci-openshift-machine-config-operator-master-e2e-gcp-op/1623093773879742464/clone-log.txt